database.hh

```diff
@@ -1482,7 +1482,17 @@ public:
     future<mutation> apply_counter_update(schema_ptr, const frozen_mutation& m, db::timeout_clock::time_point timeout, tracing::trace_state_ptr trace_state);
     keyspace::config make_keyspace_config(const keyspace_metadata& ksm);
     const sstring& get_snitch_name() const;
-    future<> clear_snapshot(sstring tag, std::vector<sstring> keyspace_names);
+    /*!
+     * \brief clear snapshot based on a tag
+     * The clear_snapshot method deletes specific or multiple snapshot
+     * You can specify:
+     * tag - The snapshot tag (the one that was used when creating the snapshot) if not specify
+     *       All snapshot will be deleted
+     * keyspace_names - a vector of keyspace names that will be deleted, if empty all keyspace
+     *                  will be deleted.
+     * table_name - A name of a specific table inside the keyspace, it empty all tables will be deleted.
+     */
+    future<> clear_snapshot(sstring tag, std::vector<sstring> keyspace_names, const sstring& table_name = "");
```

Comment:
default parameters were discovered to cause a couple of nasty bugs lately, maybe it's acceptable to drop it? This function only has 1 caller anyway, so it's not much work to pass an additional parameter (preferably `std::nullopt`, after transforming this const reference to std::optional`.

Reply:
what make your argument even stronger is the API always supplies that extra parameter and the only reason to make it optional is that I can break it into different patches.
I can still do that with a minimal change on the API call in one patch, and the rest in another patch